src/tools/cephfs/first-damage.py

```diff
@@ -87,7 +87,7 @@ def traverse(MEMO, ioctx):
                     log.debug(f'\t{dnk}: val size {len(val)}')
                     (first,) = struct.unpack('<Q', val[:8])
                     log.debug(f'\t{dnk}: first {first}')
-                    if first > NEXT_SNAP:
+                    if first > NEXT_SNAP or first == CEPH_NOSNAP:
```

Comment:
shouldn't `first > NEXT_SNAP` capture this? I believe that's why I didn't explicitly test this.

Reply:
If this change looks fine, I'll merge this since @kotreshhr is on PTO this week and follow up with the hunk removed. @batrick

Reply:
We also need to update `pack_into` below. It using `<I` instead of `<Q`.

Reply:
https://docs.python.org/3/library/struct.html#format-characters